### PR TITLE
Extract pod metrics and cluster frame context

### PR DESCRIPTION
**Pull request by samitiilikainen**

### CODE REVIEW

File: packages/cluster-frame-context/index.ts [hunk 1]
```diff
@@ -14,3 +16,7 @@ export interface ClusterContext {
   isLoadingAll(namespaces: string[]): boolean;
   isGlobalWatchEnabled(): boolean;
 }
+
+export const clusterFrameContextForClusterScopedResourcesInjectionToken = getInjectionToken<ClusterContext>({
+  id: "cluster-frame-context-for-cluster-scoped-resources-injection-token",
+});
```
Comment (Nokel81) [inline packages/cluster-frame-context/index.ts]: Can you also export an injection token for the NamespaceScoped resources too?

File: packages/cluster-frame-context/index.ts [hunk 2]
```diff
@@ -2,10 +2,12 @@
  * Copyright (c) OpenLens Authors. All rights reserved.
  * Licensed under MIT License. See LICENSE in root directory for more information.
  */
-
 /**
- * This type is used for KubeObjectStores
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
  */
```
Comment (Nokel81) [inline packages/cluster-frame-context/index.ts]: duplicate header

File: packages/cluster-frame-context/index.ts
Comment (Nokel81) [inline packages/cluster-frame-context/index.ts:1]: Not about this file in general. Might be better if we also had a `src/` which this just exported from

File: packages/utility-features/kube-api/src/endpoints/index.ts [hunk 3]
```diff
@@ -48,3 +48,4 @@ export * from "./service-account.api";
 export * from "./stateful-set.api";
 export * from "./storage-class.api";
 export * from "./vertical-pod-autoscaler.api";
+export { podMetricsApiInjectionToken } from "./pod-metrics-api-injection-token";
```
Comment (Nokel81) [inline packages/utility-features/kube-api/src/endpoints/index.ts]: This is not the place for this, see my PR #7792

**Review by Nokel81 — CHANGES_REQUESTED** [review]

File: packages/cluster-frame-context/index.ts
Comment (samitiilikainen) [reply packages/cluster-frame-context/index.ts:1]: Yeah, created src

File: packages/cluster-frame-context/index.ts (hunk 2 above)
Comment (samitiilikainen) [reply packages/cluster-frame-context/index.ts]: Thanks, fixed

File: packages/utility-features/kube-api/src/endpoints/index.ts (hunk 3 above)
Comment (samitiilikainen) [reply packages/utility-features/kube-api/src/endpoints/index.ts]: Thanks, rebased to get #7792 and while resolving conflicts also did this move

**Pull request closed without merging**